lib/samples/control_annotation_sublayer_visibility/control_annotation_sublayer_visibility.dart

```diff
@@ -0,0 +1,249 @@
+              visible: !_ready,
+              child: const SizedBox.expand(
+                child: ColoredBox(
+                  color: Colors.white30,
+                  child: Center(child: CircularProgressIndicator()),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+      bottomSheet: _settingsVisible ? buildSettings(context) : null,
+    );
+  }
+
+  // The build method for the Settings bottom sheet.
+  Widget buildSettings(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.fromLTRB(
+        20.0,
+        20.0,
+        20.0,
+        max(
+          20.0,
+          View.of(context).viewPadding.bottom /
+              View.of(context).devicePixelRatio,
+        ),
+      ),
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Row(
+            children: [
+              Text(
+                'Settings',
+                style: Theme.of(context).textTheme.titleLarge,
+              ),
+              const Spacer(),
+              IconButton(
+                icon: const Icon(Icons.close),
+                onPressed: () => setState(() => _settingsVisible = false),
+              ),
+            ],
+          ),
+          Row(
+            children: [
+              Text(
+                _openLabel,
+                style: TextStyle(color: _openLabelColor),
+              ),
+              const Spacer(),
+              Switch(
+                value: _openSublayer.isVisible,
+                onChanged: (value) {
+                  // Set the visibility of the open sub layer.
+                  setState(() => _openSublayer.isVisible = value);
+                },
+              ),
+            ],
+          ),
+          Row(
+            children: [
+              Text(
+                _closedLabel,
+              ),
+              const Spacer(),
+              Switch(
+                value: _closedSublayer.isVisible,
+                onChanged: (value) {
+                  // Set the visibility of the closed sub layer.
+                  setState(() => _closedSublayer.isVisible = value);
+                },
+              ),
+            ],
+          ),
+          Text(
+            _currentScaleLabel,
+          ),
+        ],
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    try {
+      await downloadSampleData(['b87307dcfb26411eb2e92e1627cb615b']);
+      final appDir = await getApplicationDocumentsDirectory();
+
+      // Load the mobile map package.
+      final mmpkFile = File('${appDir.absolute.path}/GasDeviceAnno.mmpk');
+      // Mobile map package that contains annotation layers.
+      final mmpk = MobileMapPackage.withFileUri(mmpkFile.uri);
+      await mmpk.load();
+
+      if (mmpk.maps.isNotEmpty) {
+        // Get the first map in the mobile map package and set to the map view.
+        _mapViewController.arcGISMap = mmpk.maps.first;
+      }
+
+      if (_mapViewController.arcGISMap != null) {
```

Comment:
Likewise we no longer need to check arcGISMap for null -- we know it has been set.